src/cliapp.rs

```diff
@@ -26,10 +26,94 @@ pub fn make_app() -> App<'static, 'static> {
                 .global(true)
                 .help("The path to the config folder."),
         )
-        .subcommand(App::new("run").about("Run the relay").after_help(
-            "This runs the relay in the foreground until it's shut down.  It will bind \
+        .subcommand(
+            App::new("run")
+                .about("Run the relay")
+                .after_help(
+                    "This runs the relay in the foreground until it's shut down.  It will bind \
              to the port and network interface configured in the config file.",
-        ))
+                )
+                .arg(
+                    Arg::with_name("secret_key")
+                        .long("secret-key")
+                        .short("s")
+                        .value_name("KEY")
+                        .requires("public_key")
+                        .help("The secret key to set"),
+                )
+                .arg(
+                    Arg::with_name("public_key")
+                        .long("public-key")
+                        .short("p")
+                        .value_name("KEY")
+                        .requires("secret_key")
+                        .help("The public key to set"),
+                )
+                .arg(
+                    Arg::with_name("id")
+                        .long("id")
+                        .short("i")
+                        .value_name("RELAY_ID")
+                        .help("The relay ID to set"),
+                )
+                .arg(
+                    Arg::with_name("upstream-url")
+                        .value_name("UPSTREAM_URL")
+                        .takes_value(true)
+                        .short("u")
+                        .long("upstream-url")
+                        .help("The upstream server URL."),
+                )
+                .arg(
+                    Arg::with_name("host")
+                        .value_name("HOST")
+                        .takes_value(true)
+                        .short("H")
+                        .long("host")
+                        .help("The host dns name."),
+                )
+                .arg(
+                    Arg::with_name("port")
+                        .value_name("PORT")
+                        .takes_value(true)
+                        .short("P")
+                        .long("port")
+                        .help("The server port."),
+                )
+                .arg(
+                    Arg::with_name("processing-enabled")
+                        .short("e")
+                        .long("processing-enabled")
+                        .help("Enable processing."),
+                )
+                .arg(
+                    Arg::with_name("processing-disabled")
```

Comment:
Since this is the default, I'd remove this. 

If you strongly prefer to keep it, please add `.conflicts_with("processing_enabled")`

Reply:
see comment above about my preference. Note that there are already mutually exclusive since tye are in an arg group:

                .arg(
                    Arg::with_name("processing-enabled")
                        .short("e")
                        .long("processing-enabled")
                        .help("Enable processing."),
                )
                .arg(
                    Arg::with_name("processing-disabled")
                        .short("d")
                        .long("processing-disabled")
                        .help("Disable processing."),
                )
                .group(
                    ArgGroup::with_name("processing")
                        .args(&["processing-enabled", "processing-disabled"])
                        .multiple(false),
                )